target/ppc/int_helper.c

```diff
@@ -1529,56 +1529,49 @@ VEXTU_X_DO(vextuwrx, 32, 0)
-    } else {                                                                \
-        int len_a = array_size - ofs, len_b = sz - len_a;                   \
-                                                                            \
-        memmove(&tmp.u8[array_size / 2 - sz], &a->u8[array_size - len_a],   \
-                len_a);                                                     \
-        memmove(&tmp.u8[array_size / 2 - len_b], &b->u8[0], len_b);         \
-    }                                                                       \
-                                                                            \
-    *t = tmp;                                                               \
-}
+#define ELEM_ADDR(VEC, IDX, SIZE) (&(VEC)->VsrB(IDX))
 #else
-#define VEXTDVLX(name, sz) \
-void glue(glue(helper_VEXTD, name), VLX)(ppc_avr_t *t, ppc_avr_t *a,        \
-                                         ppc_avr_t *b, target_ulong ofs)    \
-{                                                                           \
-    int array_size = ARRAY_SIZE(t->u8);                                     \
-    ppc_avr_t tmp = { .u64 = { 0, 0 } };                                    \
-    ofs = ofs & 0x1F;                                                       \
-                                                                            \
-    if (ofs + sz <= array_size) {                                           \
-        memmove(&tmp.u8[array_size / 2], &a->u8[array_size - sz - ofs], sz);\
-    } else if (ofs >= array_size) {                                         \
-        ofs -= array_size;                                                  \
-        memmove(&tmp.u8[array_size / 2], &b->u8[array_size - sz - ofs], sz);\
-    } else {                                                                \
-        int len_a = array_size - ofs, len_b = sz - len_a;                   \
-                                                                            \
-        memmove(&tmp.u8[array_size / 2], &b->u8[array_size - len_b], len_b);\
-        memmove(&tmp.u8[array_size / 2 + len_b], &a->u8[0], len_a);         \
-    }                                                                       \
-                                                                            \
-    *t = tmp;                                                               \
-}
+#define ELEM_ADDR(VEC, IDX, SIZE) (&(VEC)->VsrB(IDX) - (SIZE) + 1)
 #endif
-VEXTDVLX(UB, 1)
-VEXTDVLX(UH, 2)
-VEXTDVLX(UW, 4)
-VEXTDVLX(D, 8)
+#define VEXTDVLX(NAME, TYPE) \
+void glue(glue(helper_VEXTD, NAME), VLX)(CPUPPCState *env, ppc_avr_t *t,       \
+                                         ppc_avr_t *a, ppc_avr_t *b,           \
+                                         target_ulong index)                   \
+{                                                                              \
+    int array_size = ARRAY_SIZE(t->u8), elem_size = sizeof(TYPE);              \
```

Comment:
Probably doesn't make any real difference (except maybe for people reading the code), but I would make these 'const int'.